CMakeLists.txt

```
@@ -1,26 +1,112 @@
 cmake_minimum_required(VERSION 3.10)
-project(vikunja)
+project(vikunja
+  VERSION      0.1.0
+  DESCRIPTION  "Primitives for Alpaka"
+  HOMEPAGE_URL "https://github.com/ComputationalRadiationPhysics/vikunja"
+  LANGUAGES    CXX)
 
 set(CMAKE_CXX_STANDARD 14)
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra")
+set(CMAKE_CXX_STANDARD_REQUIRED 14)
+
+option(vikunja_BUILD_EXAMPLES "Build the examples" OFF)
+option(vikunja_BUILD_DOXYGEN "Build the doxygen documentation" OFF)
+option(vikunja_ENABLE_EXTRA_WARING "Enable extra warnings" ON)
+option(BUILD_TESTING "Build the testing tree." OFF)
```

Suggested change:

```suggestion
option(BUILD_TESTING "Build the testing tree." OFF)
include(CTest)
```

CTest listens to the `BUILD_TESTING` option (defaults to `ON`, so we still need to override) and gives us the nice `ctest` command line interface. Since we already use `add_test` in the `test` directory this comes without additional work.